docs/guides/upgrades/v0.38-to-v1.0.md

```diff
@@ -0,0 +1,598 @@
+support will be dropped in future releases and users are encouraged to use the versioned
+approach. For example, instead of `curl localhost:26657/block?height=5`,
+use `curl localhost:26657/v1/block?height=5`.
+
+### Websocket endpoint
+
+The `/websocket` endpoint path is no longer configurable in the client or server.
+Creating an RPC client now takes the form:
+
+```golang
+// The WebSocket endpoint in the following example is assumed to be available
+// at http://localhost:26657/v1/websocket
+rpcClient, err := client.New("http://localhost:26657/v1")
+```
+
+For more information please see this [PR](https://github.com/cometbft/cometbft/pull/1412)
+
+## Go API Surface Area
+
+### Package Internalization
+
+As per [ADR 109](https://staging-cometbft-docs.netlify.app/staging/references/architecture/adr-109-reduce-go-api-surface.md), the
+following packages that were publicly accessible in CometBFT `v0.38.x` were moved
+into the [internal](https://github.com/cometbft/cometbft/tree/main/internal) directory:
+
+- `blocksync`
+- `consensus`
+- `evidence`
+- `inspect`
+- `libs/async`
+- `libs/autofile`
+- `libs/bits`
+- `libs/clist`
+- `libs/cmap`
+- `libs/events`
+- `libs/fail`
+- `libs/flowrate`
+- `libs/net`
+- `libs/os`
+- `libs/progressbar`
+- `libs/rand`
+- `libs/strings`
+- `libs/tempfile`
+- `libs/timer`
+
+If you depend on any of these packages and wish for them to be made public again, please [submit an issue](https://github.com/cometbft/cometbft/issues/new/choose)
+outlining your use case. We will then assess the most effective way to assist you.
+
+## Legacy gRPC
+
+### Broadcast API removed
+
+CometBFT `v0.38.x` offered a simplistic gRPC support with only one method equivalent to the `/broadcast_tx_commit` endpoint.
+This has been [removed](https://github.com/cometbft/cometbft/pull/659) from CometBFT `v1.0` in favor of
+the new [Data Companion gRPC services](https://docs.cometbft.com/v1.0/explanation/data-companion/grpc).
+
+The following configuration parameters in `config.toml` are not valid anymore in `v1.0`:
+
+```toml
+[rpc]
+
+grpc_laddr = ""
+grpc_max_open_connections = 900
+```
+
+## Data Companion API
+
+CometBFT `v1.0` introduces support for a new Data Companion Pull API as specified in
+[ADR-101](https://github.com/cometbft/cometbft/blob/main/docs/references/architecture/adr-101-data-companion-pull-api.md)
+
+### gRPC services
+
+There is a whole new [section](#grpc-section) in the `config.toml` to support the new Data Companion gRPC services.
+
+In the CometBFT `v1.0` configuration file, if the `laddr` parameter in the `[grpc]` section is not specified,
+all the services will be **disabled**. If an address is specified, the node operator can selectively choose
+which services individually should be enabled or disabled.
+
+### Pruning service
+
+As part of the new Data Companion API, a new [Pruning Service](https://docs.cometbft.com/v1.0/explanation/data-companion/pruning) is available which can be used to influence
```

Comment:
I would rewrite this to clarify that the pruning service is responsible for deleting past data (older than N blocks).